### PR TITLE
Improve accessibility according to web guidelines

### LINKED ISSUE #25: Make Foundation compatible with Webrichtlijnen
In gitlab by dorien on Nov 26, 2015, 11:20

http://versie2.webrichtlijnen.nl/

Kunnen we dan voldoen aan keurmerk http://www.drempelvrij.nl

### PR DESCRIPTION
**Pull request by ddeboer**
Fix #25.

### CODE REVIEW

File: modules/mod_ginger_base/templates/logo/logo.tpl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% block button %}
     <a href="/" class="logo {{ extraClasses }}">
-        <img src="/lib/images/logo.png">
+        <img src="/lib/images/logo.png" alt="dit is het logo van de specifieke site">
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/logo/logo.tpl]: Rather a dynamic value such as `{{ m.config.site.title.value }}`

File: modules/mod_ginger_base/templates/search-suggestions/search-form.tpl [hunk 2]
```diff
@@ -21,9 +21,11 @@ as
 
 <form class="{{ formclass }} {{ extraFormClassess }}" id="search-suggestions__searchform-{{ #identifier }}" role="search" action="{% if context %}/{{ context }}_search{% else %}{% url search %}{% endif %}" method="get">
     <div class="{{ wrapperclass }}">
+        <label for="qs">Zoeken</label>
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/search-suggestions/search-form.tpl]: Even though only used for screen readers, all labels must be in English: `{_ Search _}`.

File: modules/mod_ginger_base/templates/head/favicon.tpl [hunk 3]
```diff
@@ -1,5 +1,5 @@
-<link rel="apple-touch-icon-precomposed" sizes="144x144" href="/lib/images/apple-touch-icon-144x144.png" />
-<link rel="apple-touch-icon-precomposed" sizes="152x152" href="/lib/images/apple-touch-icon-152x152.png" />
+<link rel="apple-touch-icon" sizes="144x144" href="/lib/images/apple-touch-icon-144x144.png" />
+<link rel="apple-touch-icon" sizes="152x152" href="/lib/images/apple-touch-icon-152x152.png" />
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/head/favicon.tpl]: @DorienD @Devenwel @lvandergrient Will this cause any problems?

File: modules/mod_ginger_base/templates/search-suggestions/search-form.tpl [hunk 4]
```diff
@@ -21,9 +21,11 @@ as
 
 <form class="{{ formclass }} {{ extraFormClassess }}" id="search-suggestions__searchform-{{ #identifier }}" role="search" action="{% if context %}/{{ context }}_search{% else %}{% url search %}{% endif %}" method="get">
     <div class="{{ wrapperclass }}">
+        <label class="search-form__label" for="qs">Zoeken</label>
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/search-suggestions/search-form.tpl]: `{_Search_}` for label

File: modules/mod_ginger_base/templates/head/favicon.tpl (hunk 3 above)
Comment (DorienD) [inline modules/mod_ginger_base/templates/head/favicon.tpl]: Nope not at all, precomposed means that it won't get the styles apple puts around it automatically. So no problems here.

Comment (ddeboer) [inline modules/mod_ginger_base/templates/head/favicon.tpl]: But `precomposed` was *removed*. So does that mean Apple styling applies and may cause a visual change?

File: modules/mod_ginger_base/templates/media/media.image.tpl [hunk 5]
```diff
@@ -15,13 +15,13 @@
         {% endif %}
 
             {% if id.medium.width > 750 %}
-                {% image id.id mediaclass="landscape-"++sizename alt="" crop=id.crop_center %}
+                {% image id.id mediaclass="landscape-"++sizename alt="landscape-"++sizename crop=id.crop_center %}
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/media/media.image.tpl]: Does `id.title` give a better `alt` value?

File: modules/mod_ginger_base/templates/head/favicon.tpl (hunk 3 above)
Comment (DorienD) [inline modules/mod_ginger_base/templates/head/favicon.tpl]: I've tested it on my iphone, and there is no visual change. The precomposed was for the purpose of ignoring the styling of ios (like shiny stuff and shadows) but i think this is no longer added by apple. See screenshot for test.

img_2797

File: modules/mod_ginger_base/templates/media/media.image.tpl (hunk 5 above)
Comment (yorivanloo) [inline modules/mod_ginger_base/templates/media/media.image.tpl]: yes agree

File: modules/mod_ginger_base/templates/media/media.image.tpl [hunk 6]
```diff
@@ -15,13 +15,13 @@
         {% endif %}
 
             {% if id.medium.width > 750 %}
-                {% image id.id mediaclass="landscape-"++sizename alt="" crop=id.crop_center %}
+                {% image id.id mediaclass="landscape-"++sizename alt=id.title crop=id.crop_center %}
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/media/media.image.tpl]: Title can contain spaces so needs to be wrapped in `"`

Comment (ddeboer) [inline modules/mod_ginger_base/templates/media/media.image.tpl]: And of course this change needs to be applied for all `alt`s.

**Comment by ddeboer** [comment]
@yowri We still have the mediaclasses as alt titles.

File: modules/mod_ginger_base/templates/list/list-item-vertical.tpl [hunk 7]
```diff
@@ -3,13 +3,12 @@
...
 
 {% if id.is_visible %}
-
     <li class="{% block class %} list__item--vertical{% endblock %} {{ extraClasses }}">
 
         <a href="{{ id.page_url }}">
             <article class="cf">
                 <div class="list__item--vertical__image" {% if dep_rsc %} style="background-image: url({% image_url dep_rsc.id mediaclass="list-image" %});"{% endif %}>
-                    {% image dep_rsc.id mediaclass="list-image" class="list__item__image" alt="" title="" crop=dep_rsc.crop_center %}
+                    {% image dep_rsc.id mediaclass="list-image" class="list__item__image" alt=id.title title="" crop=dep_rsc.crop_center %}
```
Comment (ddeboer) [inline modules/mod_ginger_base/templates/list/list-item-vertical.tpl]: Needs `"`

**Pull request merged**